### DISCUSSION TITLE
The fields part, supplier, SKU must make a unique set - include manufacturer part into constraint?

### DISCUSSION THREAD

**Discussion by turist79** [inventree/InvenTree#8583] [Q&A] [answered]
We have situation, where for one our part there are different manufacturer parts supplied by the same supplier under same sku.
So the combination: **part + manufacturer part + supplier, sku** is unique, but it does not allow to save the record with abovementioned error.

As i understand, the problem here is that "part, supplier, SKU" is a constraint and adding manufacturer part will mean that it won't be able to be null, otherwise the whole thing wont work?
If so, isn`t there a workaround for that, for example discussed here https://stackoverflow.com/questions/8289100/create-unique-constraint-with-null-columns ?

**Comment by SchrodingersGat** [comment, marked as answer]
> where for one our part there are different manufacturer parts supplied by the same supplier under same sku.

Surely the point of a supplier SKU is that is unique, and can be used to guarantee supply of a very specific part.

I would suggest that in this case you should add a prefix or suffix to the SKU - to keep them unique - and to differentiate them. I do not think that we should remove this uniqueness requirement to appease this edge case

**Reply by turist79** [reply]
> > where for one our part there are different manufacturer parts supplied by the same supplier under same sku.
> 
> Surely the point of a supplier SKU is that is unique, and can be used to guarantee supply of a very specific part.
> 
> I would suggest that in this case you should add a prefix or suffix to the SKU - to keep them unique - and to differentiate them. I do not think that we should remove this uniqueness requirement to appease this edge case

adding prefix or suffix was my 1st idea of course.
I was not suggesting to lift the uniqueness requirement though, but to include manufacturer part as part of the constraint. If the underlying database or the framework itself is able to work around the case when manufacturer part is not specified (so the field set is part-null (as manufacturer part) - supplier - sku), then it won't alter the status quo.
It will just add the possibility to keep track of additional info about actual manufacturer part supplied by the supplier, if it is applicable (for example, we transfer not only supplier's sku number, but really the combination of his SKU and full manufacturer`s number llike some sort of revision number making the combination unique)

PS: i assume that it may be not applicable for electronic components/production but is perfectly fine in many other areas.

Anyway, of course i respect your vision of the app architecture.
It is more a question of convenience for us, not big of an issue.
